test/Dialect/FIRRTL/lower-layers.mlir

```diff
@@ -116,3 +117,69 @@ firrtl.circuit "ModuleNameConflict" {
 // CHECK:         firrtl.instance foo @ModuleNameConflict_A()
 // CHECK-NEXT:    firrtl.instance {{[_a-zA-Z0-9]+}} {lowerToBind,
 // CHECK-SAME:      @[[groupModule]](
+
+// -----
+
+firrtl.circuit "Test" {
+  firrtl.module @Test() {}
+
+  firrtl.layer @A bind {
+    firrtl.layer @B bind {
+      firrtl.layer @C bind {}
+    }
+  }
+  firrtl.layer @B bind {}
+
+  //===--------------------------------------------------------------------===//
+  // Removal of Probe Colors
+  //===--------------------------------------------------------------------===//
+
+  // CHECK-LABEL: @ColoredPorts(out %o: !firrtl.probe<uint<1>>)
+  firrtl.module @ColoredPorts(out %o: !firrtl.probe<uint<1>, @A>) {}
+
+  // CHECK-LABEL: @ExtColoredPorts(out o: !firrtl.probe<uint<1>>)
+  firrtl.extmodule @ExtColoredPorts(out o: !firrtl.probe<uint<1>, @A>)
+
+  // CHECK-LABEL: @ColoredPortsOnInstances
+  firrtl.module @ColoredPortsOnInstances() {
+    // CHECK: %foo_o = firrtl.instance foo @ColoredPorts(out o: !firrtl.probe<uint<1>>)
+   %foo_o = firrtl.instance foo @ColoredPorts(out o: !firrtl.probe<uint<1>, @A>)
+  }
+
+  // CHECK-LABEL: @ColoredThings
+  firrtl.module @ColoredThings() {
+    // CHECK: %0 = firrtl.wire : !firrtl.probe<bundle<f: uint<1>>>
+    %0 = firrtl.wire : !firrtl.probe<bundle<f: uint<1>>, @A>
+    // CHECK: %1 = firrtl.ref.sub %0[0] : !firrtl.probe<bundle<f: uint<1>>>
+    %1 = firrtl.ref.sub %0[0] : !firrtl.probe<bundle<f: uint<1>>, @A>
+    // CHECK-NOT: firrtl.cast
+    %2 = firrtl.ref.cast %1 : (!firrtl.probe<uint<1>, @A>) -> !firrtl.probe<uint<1>, @A::@B>
+  }
+
+  //===--------------------------------------------------------------------===//
+  // Removal of Enabled Layers
+  //===--------------------------------------------------------------------===//
+
+  // CHECK-LABEL: @EnabledLayers() {
+  firrtl.module @EnabledLayers() attributes {layers = [@A]} {}
+
+  // CHECK-LABEL: @EnabledLayersOnInstance()
+  firrtl.module @EnabledLayersOnInstance() attributes {layers = [@A]} {
+    // CHECK: firrtl.instance enabledLayers @EnabledLayers()
+    firrtl.instance enabledLayers {layers = [@A]} @EnabledLayers()
+  }
+
+  //===--------------------------------------------------------------------===//
+  // Removal of Layerblocks and Layers
+  //===--------------------------------------------------------------------===//
+
+  // CHECK-NOT: firrtl.layer @GoodbyeCruelWorld
+  firrtl.layer @GoodbyeCruelWorld bind {}
+
+  // CHECK-LABEL @WithLayerBlock
+  firrtl.module @WithLayerBlock() {
+    // CHECK-NOT firrtl.layerblock @GoodbyeCruelWorld
+    firrtl.layerblock @GoodbyeCruelWorld {
+    }
+  }
+}
```

Comment:
The main thing is this needs some test like `Simple` that is testing probe color logic other than only removal. 

Also this needs probably two end-to-end `firtool` tests that are checking FIRRTL to Verilog.